src/mlpack/methods/ann/layer/add_merge.hpp

```diff
@@ -107,15 +114,33 @@ class AddMerge
   //! Modify the delta.
   OutputDataType& Delta() { return delta; }
 
+  //! Return the model modules.
+  // std::vector<LayerTypes<CustomLayers...> >& Model()
+  // {
+  //   if (model)
+  //   {
+  //     return network;
+  //   }
+
+  //   return empty;
+  // }
+
   /**
    * Serialize the layer.
    */
   template<typename Archive>
   void serialize(Archive& ar, const unsigned int /* version */);
 
  private:
+  //! Parameter which indicates if the modules should be exposed.
+  bool model;
```

Comment:
Hm, I guess I am a little confused about `model`---if it is `true`, then we delete the layers in the destructor.  Maybe a better name is `ownsLayers` or something?  I am not sure if I have understood fully so correct me if I'm wrong.

Reply:
Good point, changed to ``ownsLayers``.